models/goodsDao.js

```diff
@@ -0,0 +1,47 @@
+const appDataSource = require("./dataSource");
+
+const goodsDao = async(category_id) => {
+    try{
+        let result = ``;
+        if(Number(category_id) <=9 && Number(category_id) >= 1){
+            result = await appDataSource.query(
+                `
+                SELECT *
+                FROM products
+                WHERE products.sub_category_id = ?
+                `,[category_id]
+            
+                )
+            }else{
+                const error = new Error('INVALID_NUMBER_INPUT');
+                error.statusCode = 400;
+                throw error
+                
+        }
```

Comment:
- 상기 코드를 아래와 같이 깔끔하게 수정해주세요!
- category_id에 대한 예외 처리를 DAO에서 하게 된다면, 추후 확장성이 확보가 되지 않는 코드처럼 보여집니다.
    - 언제든 카테고리 아이디가 늘어날 수도 있고, 때로는 변경될 수도 있기 때문이지요!
- 즉 데이터베이스내 없는 카테고리 pk 값에 대한 select문을 호출한다면, 결과론적으로 빈배열 등이 나올 예정입니다.
- 작성해주신것처럼 사전 예외처리가 아닌 해당 내용이 빈배열로 나왔을 때의 사후 예외처리로서 접근하시는 것을 권장드립니다.

```suggestion
const goodsDao = async(category_id) => {
    try{
            result = await appDataSource.query(
                `
                SELECT 
                   id,
                   name,
                   ... 이하 생략
                FROM products
                WHERE products.sub_category_id = ?
                `,[category_id]
                )

```